.github/workflows/build_pr_documentation.yml

```diff
@@ -18,20 +19,18 @@ jobs:
       PR_CLONE_URL: ${{ github.event.pull_request.head.repo.clone_url }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
-          repository: 'huggingface/doc-builder'
+          repository: "huggingface/doc-builder"
           path: doc-builder
 
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
-          repository: 'huggingface/optimum-intel'
+          repository: "huggingface/optimum-intel"
           path: optimum-intel
 
       - name: Setup environment
         run: |
-          python -m venv venv-doc
-          source venv-doc/bin/activate
```

Comment:
This was needed when `ubuntu-latest` was upgraded to `ubuntu-24.04` couple of weeks ago (now reverted) which uses python 3.12. If we want to remove it, we need to replace `ubuntu-latest` with `ubuntu-22.04` to avoid future issues